aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/DeleteHandler.java

```diff
@@ -40,25 +39,23 @@ public DeleteHandler(final HandlerConfig config) {
     ) {
         final ResourceModel model = request.getDesiredResourceState();
 
-        if (!callbackContext.isDeleting()) {
-            boolean deletionProtectionEnabled;
-            try {
-                deletionProtectionEnabled = isDeletionProtectionEnabled(proxyClient, model);
-            } catch (Exception exception) {
-                return Commons.handleException(
-                        ProgressEvent.progress(model, callbackContext),
-                        exception,
-                        DEFAULT_DB_CLUSTER_ERROR_RULE_SET
-                );
-            }
-            if (deletionProtectionEnabled) {
-                return ProgressEvent.failed(
-                        model,
-                        callbackContext,
-                        HandlerErrorCode.NotUpdatable,
-                        String.format(DELETION_PROTECTION_ENABLED_ERROR, model.getDBClusterIdentifier())
-                );
-            }
+        boolean deletionProtectionEnabled;
+        try {
+            deletionProtectionEnabled = isDeletionProtectionEnabled(proxyClient, model);
+        } catch (Exception exception) {
+            return Commons.handleException(
+                    ProgressEvent.progress(model, callbackContext),
+                    exception,
+                    DEFAULT_DB_CLUSTER_ERROR_RULE_SET
+            );
+        }
+        if (deletionProtectionEnabled) {
+            return ProgressEvent.failed(
+                    model,
+                    callbackContext,
+                    HandlerErrorCode.NotUpdatable,
+                    String.format(DELETION_PROTECTION_ENABLED_ERROR, model.getDBClusterIdentifier())
+            );
```

Comment:
Nit: Should deletionProtectionEnabled be a chain step that return progress if it is not enabled. 
If it is enable then return failed progress.
```suggestion
        return ProgressEvent.progress(model, callbackContext)
                .then(progress -> checkDeletionProtection(proxy, request, proxyClient, progress))
                .then(progress -> removeFromGlobalClusterIfMember(proxy, proxyClient, progress, model.getGlobalClusterIdentifier());)
                .then(progress -> deleteDbCluster(proxy, request, proxyClient, progress));

Reply:
@moataz-mhmd it's a good idea, will re-submit the PR soon. Thank you!